vmdb/lib/miq_automation_engine/engine/miq_ae_state_machine.rb

```diff
@@ -21,20 +21,22 @@ def increment_state_retries
     end
 
     def enforce_state_maxima(f)
-      now = Time.zone.now.utc
+      enforce_max_retries(f)
+      enforce_max_time(f)
+    end
 
-      unless f['max_retries'].blank?
-        if @workspace.root['ae_state_retries'].to_i > f['max_retries'].to_i
-          raise "number of retries <#{@workspace.root['ae_state_retries']}> exceeded maximum of <#{f['max_retries']}>"
-        end
-      end
+    def enforce_max_retries(f)
+      return if f['max_retries'].blank?
+      return if f['max_retries'].to_i >= @workspace.root['ae_state_retries'].to_i
+      raise "number of retries <#{@workspace.root['ae_state_retries']}> exceeded maximum of <#{f['max_retries']}>"
+    end
 
-      unless f['max_time'].blank?
-        ae_state_started = Time.zone.parse(@workspace.root['ae_state_started'])
-        if ae_state_started + f['max_time'].to_i_with_method <= now
-          raise "time in state <#{now - ae_state_started} seconds> exceeded maximum of <#{f['max_time']}>"
-        end
-      end
+     
+    def enforce_max_time(f)
+      return if f['max_time'].blank?
+      lapsed = Time.zone.now.utc - Time.zone.parse(@workspace.root['ae_state_started'])
```

Comment:
@tenderlove does the `.utc` actually do anything here for date math between 2 different `TimeWithZone` objects?